### PR TITLE
MM-14691 Add channel header message for SurveyBot

### PR DESCRIPTION
**Pull request by hmhealey**
Using the hook lets us set the header regardless of who how the DM channel was created.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-14691

### CODE REVIEW

File: server/hooks.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	// Set the description for any DM channels opened between Surveybot and a user
+	if channel.Type != model.CHANNEL_DIRECT {
+		return
+	}
+
+	if !strings.HasPrefix(channel.Name, p.botUserId+"__") && !strings.HasSuffix(channel.Name, "__"+p.botUserId) {
+		return
+	}
+
+	channel.Header = "Surveybot collects user feedback to improve Mattermost. [Learn more](https://mattermost.com/pl/default-nps)."
```
Comment (saturninoabril) [inline server/hooks.go]: Page is not yet available for this URL but I guess it will be to follow.

Comment (hmhealey) [reply server/hooks.go]: Yep, there's a separate ticket to go set that up once the docs are finished

Comment (hanzei) [reply server/hooks.go]: :+1: for using a permalink

File: server/hooks.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+func (p *Plugin) ChannelHasBeenCreated(c *plugin.Context, channel *model.Channel) {
+	// Set the description for any DM channels opened between Surveybot and a user
+	if channel.Type != model.CHANNEL_DIRECT {
+		return
+	}
+
+	if !strings.HasPrefix(channel.Name, p.botUserId+"__") && !strings.HasSuffix(channel.Name, "__"+p.botUserId) {
```
Comment (lieut-data) [inline server/hooks.go]: This seems fine -- given the knowledge of how we encode direct channel names -- but would it worth querying for the users and explicitly checking for `p.botUserId` for clarity in the code?

Comment (hmhealey) [reply server/hooks.go]: I don't think we need to do that. There's enough other code that relies on how these names are constructed that we'd require a lot of other changes if we were to change how those names are constructed.

**Pull request merged**